### PR TITLE
Fix/12515: Default rooms are not grayed out when creating a new workspace while offline

### PR DESCRIPTION
**Pull request by tienifr**
### Details

When offline, if the user creates a new workspace, the workspace isn't grayed out in the LHN and on the search page. This PR fixes it.
### Fixed Issues

$ https://github.com/Expensify/App/issues/12151
$ https://github.com/Expensify/App/issues/12151#issuecomment-1311445242

### Tests

1. Sign in with any account
2. Go offline
3. Click the green plus at the bottom, New workspace. Or if you already have a workspace click the avatar in the header to open the settings, click on a workspace, click the 3 vertical dots at the top, New workspace.
4. Go online and wait for all APIs successful
5. Go offline 

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Sign in with any account
2. Go offline
3. Click the green plus at the bottom, New workspace. Or if you already have a workspace click the avatar in the header to open the settings, click on a workspace, click the 3 vertical dots at the top, New workspace.
4. Verify that the default rooms are grayed out (Default rooms include - Workspace room, #admins, #announce and domain default group)
5. Go online and wait for all APIs successful
6. Go offline and verify that the default rooms aren't grayed out

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/113963320/201303589-cd0defcf-3aba-4be3-bd48-acaa35874e5d.mp4

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/113963320/202394300-c961d35e-5392-4ce2-af49-a92565f58417.mp4

#### Mobile Web - Safari

https://user-images.githubusercontent.com/113963320/202393926-914c0a8b-b294-4e3f-94d8-81e517f9e6a5.mp4

#### Desktop

https://user-images.githubusercontent.com/113963320/202393767-fecdb3ae-5dfd-4f14-bc2e-28e884c9b35d.mp4

#### iOS

https://user-images.githubusercontent.com/113963320/202397810-408a9409-6c91-4d3a-b66f-abe75ed5948f.mp4

#### Android

https://user-images.githubusercontent.com/113963320/202403400-bd7e2f2a-94e4-4daf-bab0-570647faa6f8.mp4

### CODE REVIEW

**Comment by dangrous** [comment]
Adding @MonilBhavsar and @eVoloshchak to the review based on the issue, and removing myself - but if you want another pair of eyes let me know!

**Comment by tienifr** [comment]
@dangrous I am so sorry, but this PR isn't ready for review. I just marked as draft

**Comment by MonilBhavsar** [comment]
@tienifr this looks like a draft Poposal?

**Comment by MonilBhavsar** [comment]
Please only make a PR and request Pullerbear once your proposal is accepted

**Comment by tienifr** [comment]
@MonilBhavsar Thanks for your comment. I just only create a PR to make clear my proposal. Took note of your comment for the next time

**Comment by MonilBhavsar** [comment]
Thank you! Currently unassigning myself from PR Review. Feel free to comment on the issue when you are ready to submit your proposal

**Comment by eVoloshchak** [comment]
Bug: reports are greyed out when offline even after connecting back to the internet once

https://user-images.githubusercontent.com/9059945/202502532-7dadb956-640c-48c7-a405-f479bd6706c2.mov

**Comment by MonilBhavsar** [comment]
@tienifr did you get a chance to look at the above comment and why is it happening?
Meanwhile I'll take a look at he API

**Comment by tienifr** [comment]
@MonilBhavsar Yes, I'm re-checking my PR and will let you know the why it happens

**Comment by MonilBhavsar** [comment]
I had a quick look at the API and looks like this needs to be fixed in the frontend.
Hint: Please check if we're clearing pendingActions, we set in optimistic Onyx object, in the success Onyx object.

**Comment by tienifr** [comment]
Hi @eVoloshchak, I just pushed a new commit to fix the above bug. We need to update pendingFields in success data to fix it. Please help to review it. Thanks
Result:
https://user-images.githubusercontent.com/113963320/203287548-21a76776-152d-4b9e-aee4-de41e2253b4a.mp4

**Comment by MonilBhavsar** [comment]
@tienifr thanks! 
@eVoloshchak could you please take a look

**Comment by MonilBhavsar** [comment]
@tienifr could you also please add online and offline toggle steps to Test and QA section

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -786,7 +801,11 @@ function createWorkspace(ownerEmail = '', makeMeAdmin = false, policyName = '')
         {
             onyxMethod: CONST.ONYX.METHOD.MERGE,
             key: `${ONYXKEYS.COLLECTION.REPORT}${announceChatReportID}`,
-            value: {pendingAction: null},
+            value: {
```
Comment (eVoloshchak) [inline src/libs/actions/Policy.js:804]: Don't we still have to set `pendingAction` to null in all 3 of these?

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

File: src/libs/actions/Policy.js (hunk 1 above)
Comment (tienifr) [reply src/libs/actions/Policy.js:804]: @eVoloshchak I just added `pendingAction` to null, thanks for pointing out that

**Comment by eVoloshchak** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/9059945/203525043-0550e747-579b-47d8-bfc6-4864c737007d.mov




</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://user-images.githubusercontent.com/9059945/203525630-20d23bb4-14c3-4304-a041-d4cdc1279422.mp4








</details>

<details>
<summary>Mobile Web - Safari</summary>



https://user-images.githubusercontent.com/9059945/203526333-28ee3e74-8df3-497e-89c6-44efed8609e9.MP4





</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/9059945/203525083-4146be45-b360-44c3-a27a-b0ce31c65650.mov




</details>

<details>
<summary>iOS</summary>



https://user-images.githubusercontent.com/9059945/203525984-28bf8c42-5257-460e-8c9c-88e845ce70ce.MP4





</details>

<details>
<summary>Android</summary>


https://user-images.githubusercontent.com/9059945/203525804-1d18f68d-3308-4618-8095-e40aa4275799.mp4






</details>

**Review by eVoloshchak — APPROVED** [review]
LGTM!
cc: @MonilBhavsar

**Comment by tienifr** [comment]
Hi @MonilBhavsar Please help to take a look at this PR. Thanks so much

**Comment by MonilBhavsar** [comment]
Reviewing and testing now

**Review by MonilBhavsar — APPROVED** [review]
Looks good and works well!

**Comment by luacmartins** [comment]
The production deploy comment failed for this PR, but this was deployed to production on v1.2.32-2 on Nov 28.

**Pull request merged**